vizro-core/src/vizro/models/_navigation/accordion.py

```diff
@@ -79,22 +79,18 @@ def build(self, *, active_page_id=None):
             id="nav-panel",
         )
 
-    def _create_nav_links(self, pages):
-        """Creates a `NavLink` for each provided page that is registered."""
+    def _create_nav_links(self, pages: List[str]):
+        """Creates a `NavLink` for each provided page."""
         nav_links = []
+
         for page_id in pages:
-            try:
-                page = dash.page_registry[page_id]
-            except KeyError as exc:
-                raise KeyError(
-                    f"Page with ID {page_id} cannot be found. Please add the page to `Dashboard.pages`"
-                ) from exc
+            page = model_manager[ModelID(str(page_id))]
             nav_links.append(
                 dbc.NavLink(
-                    children=[page["name"]],
+                    children=page.title,
                     className="accordion-item-link",
                     active="exact",
-                    href=page["relative_path"],
+                    href=page.path,
```

Comment:
Is this required such that it can deployed elsewhere as well?

So to have the below working?
`"requests_pathname_prefix might be unset and so a relative URL like /page-2 can just be /page-2. However, when the app is deployed to a URL like /my-dash-app, then relative_path will be /my-dash-app/page-2."`

Reply:
Yes exactly! Otherwise if you deploy at `www.example.com/my-dash-app` and then click a link to `/page-2` then you'll end up at `www.example.com/page-2` rather than `www.example.com/my-dash-app/page-2`.

Reply:
@petar-qb @maxschulz-COL @nadijagraca @lingyielia - good to know ⏫